src/direction/hex_direction.rs

```diff
@@ -409,9 +416,134 @@ impl Direction {
 
     #[inline]
     #[must_use]
-    /// Returns the angle in radians of the given direction in the given `orientation`
-    pub fn angle(self, orientation: &HexOrientation) -> f32 {
-        self.angle_pointy() - orientation.angle_offset
+    /// Returns the angle in degrees of the given direction according to its `orientation`
+    ///
+    /// See [`Self::angle`] for radians angles
+    pub fn angle_degrees(self, orientation: &HexOrientation) -> f32 {
+        match orientation {
+            HexOrientation::Pointy => self.angle_pointy_degrees(),
+            HexOrientation::Flat => self.angle_flat_degrees(),
+        }
+    }
+
+    #[must_use]
+    /// Returns the direction from the given `angle` in degrees
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use hexx::*;
+    ///
+    /// let direction = Direction::from_pointy_angle_degrees(35.0);
+    /// assert_eq!(direction, Direction::Top);
+    /// ```
+    pub fn from_pointy_angle_degrees(angle: f32) -> Self {
+        Self::from_flat_angle_degrees(angle + DIRECTION_ANGLE_OFFSET_DEGREES)
+    }
+
+    #[must_use]
+    #[allow(clippy::cast_possible_truncation)]
+    /// Returns the direction from the given `angle` in degrees
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use hexx::*;
+    ///
+    /// let direction = Direction::from_flat_angle_degrees(35.0);
+    /// assert_eq!(direction, Direction::TopRight);
+    /// ```
+    pub fn from_flat_angle_degrees(angle: f32) -> Self {
+        let angle = angle % 360.0;
+        let angle = if angle < 0.0 { angle + 360.0 } else { angle };
+        let sector = (angle / DIRECTION_ANGLE_DEGREES).trunc() as i32;
+        match sector {
+            0 => Self::TopRight,
+            1 => Self::Top,
+            2 => Self::TopLeft,
+            3 => Self::BottomLeft,
+            4 => Self::Bottom,
+            _ => Self::BottomRight,
+        }
+    }
+
+    #[must_use]
+    /// Returns the direction from the given `angle` in radians
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use hexx::*;
+    ///
+    /// let direction = Direction::from_pointy_angle(0.6);
+    /// assert_eq!(direction, Direction::Top);
+    /// ```
+    pub fn from_pointy_angle(angle: f32) -> Self {
+        Self::from_flat_angle(angle + DIRECTION_ANGLE_OFFSET_RAD)
+    }
+
+    #[must_use]
+    #[allow(clippy::cast_possible_truncation)]
+    /// Returns the direction from the given `angle` in radians
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use hexx::*;
+    ///
+    /// let direction = Direction::from_flat_angle(0.6);
+    /// assert_eq!(direction, Direction::TopRight);
+    /// ```
+    pub fn from_flat_angle(angle: f32) -> Self {
+        let angle = angle % PI_2;
+        let angle = if angle < 0.0 { angle + PI_2 } else { angle };
```

Comment:
rem_euclid is probably clearer to use here, which will give you the sensible modulo behavior you need.

Reply:
Thanks !